File: agentguard/approval.py
class ApprovalStore:
    def __init__(self) -> None:
        self._decisions: dict[str, str] = {}

    def approve(self, action_id: str) -> None:
        self._decisions[action_id] = "approved"

    def deny(self, action_id: str) -> None:
        self._decisions[action_id] = "denied"

    def get(self, action_id: str) -> str | None:
        return self._decisions.get(action_id)

File: ts-agentguard/src/approval.ts
export type ApprovalDecision = "approved" | "denied";

export class ApprovalStore {
  private readonly decisions = new Map<string, ApprovalDecision>();

  approve(actionId: string): void {
    this.decisions.set(actionId, "approved");
  }

  deny(actionId: string): void {
    this.decisions.set(actionId, "denied");
  }

  get(actionId: string): ApprovalDecision | undefined {
    return this.decisions.get(actionId);
  }

  clear(): void {
    this.decisions.clear();
  }
}
